posthog/admin/tests/test_redis_mutation_admin.py

```diff
@@ -0,0 +1,324 @@
+    fieldsets = admin.get_fieldsets(request, obj=None)
+    ModelForm = admin.get_form(request, obj=None, change=False, fields=flatten_fieldsets(fieldsets))
+    form = ModelForm(request.POST, request.FILES, instance=None)
+
+    assert form.is_valid() is False
+    assert "Command is not a valid choice" in form.errors["__all__"]
+
+
+@pytest.mark.django_db
+def test_creation_of_del_redis_mutation(client: Client, superuser, redis):
+    """Test the creation of a DEL Redis mutation."""
+    client.force_login(superuser)
+
+    key = "a-del-test-key"
+    value = "a-default-value"
+    redis.set(key, value)
+
+    data = {"command": "del", "redis_key": key, "approval_threshold": 1}
+
+    response = client.post(
+        f"/admin/posthog/redismutation/add/",
+        data,
+    )
+
+    # 200 means the form is being re-displayed with errors
+    assert response.status_code == status.HTTP_302_FOUND
+
+    mutation = RedisMutation.objects.filter(redis_key=key).first()
+    assert mutation is not None
+    assert mutation.command == RedisMutation.MutationCommand.DEL
+    assert mutation.value is None
+    assert mutation.approval_threshold == 1
+    assert mutation.redis_key == key
+    assert mutation.redis_type == RedisMutation.RedisType.STRING
+    assert mutation.status == RedisMutation.Status.CREATED
+
+
+@pytest.mark.django_db
+def test_creation_of_expire_redis_mutation(client: Client, superuser, redis):
+    """Test the creation of an EXPIRE Redis mutation."""
+    client.force_login(superuser)
+
+    key = "an-expire-test-key"
+    value = "a-default-value"
+    redis.set(key, value)
+
+    data = {"command": "expire", "value": 10, "redis_key": key, "approval_threshold": 1}
+
+    response = client.post(
+        f"/admin/posthog/redismutation/add/",
+        data,
+    )
+
+    # 200 means the form is being re-displayed with errors
+    assert response.status_code == status.HTTP_302_FOUND
+
+    mutation = RedisMutation.objects.filter(redis_key=key).first()
+    assert mutation is not None
+    assert mutation.command == RedisMutation.MutationCommand.EXPIRE
+    assert int(mutation.value) == 10  # type: ignore
+    assert mutation.approval_threshold == 1
+    assert mutation.redis_key == key
+    assert mutation.redis_type == RedisMutation.RedisType.STRING
+    assert mutation.status == RedisMutation.Status.CREATED
+
+
+@pytest.mark.django_db
+def test_creation_of_set_redis_mutation(client: Client, superuser, redis):
+    """Test the creation of a SET Redis mutation."""
+    client.force_login(superuser)
+
+    key = "a-set-test-key"
+    value = "a-set-test-value"
+    redis.set(key, value)
+
+    data = {
+        "value": "anything",
+        "command": "set",
+        "redis_key": key,
+        "approval_threshold": 1,
+        # BEWARE: Failing to pass a valid JSON can raise an unrelated exception.
+        # Try re-running this test without parameters if you see some strange failure.
+        "parameters": '{"get": true}',
```

Comment:
In my case, the admin was showing an error related to being unable to find a css static file. I am not sure why that was triggered by JSON syntax errors here, but I added a warning for future test writers.